pkg/fnenv/fission/runtime.go

```diff
@@ -156,5 +156,9 @@ func createFunctionMeta(fn types.FnRef) *metav1.ObjectMeta {
 }
 
 func (fe *FunctionEnv) createRouterURL(fn types.FnRef) string {
-	return fe.routerURL + "/fission-function/" + fn.ID
+	if fn.Namespace == metav1.NamespaceDefault {
+		return fe.routerURL + "/fission-function/" + fn.ID
```

Comment:
Minor detail, just wondering, is this actually needed? Doesn't `$FISSION_ROUTER/fission-function/$FN` work the same as `$FISSION_ROUTER/fission-function/default/$FN`? If it is, we can leave out this if condition (to simplify the code). If it is not, it is a bug (or at least something that should be the case) in Fission and we should file an issue for it.

Reply:
This check is mandatory. The correct use can refer to [here](https://github.com/fission/fission/blob/master/fission/function.go#L714-L719)

I doubt that this is a bug.

Reply:
Okay, I raised an issue for it in the fission repo: https://github.com/fission/fission/issues/819 (it is not really a bug, but it is inconsistent/unexpected behavior). We can keep it like this here :)